camera/camera_balsaq.cc

```diff
@@ -8,32 +8,58 @@
 #include <iostream>
 #include <sstream>
 
-//%deps(balsa_queue)
-//%deps(message)
-
 namespace jet {
 
-constexpr double WEBCAM_EXPOSURE = 0.01;
+namespace {
+
+// Generate a config structure from a yaml config
+//
+// Asserts that the required config fields are actually set.
+// Has no defaulting behavior.
+CameraConfiguration generate_capture_config(const Config& config) {
+  assert(config["frames_per_second"]);
+  assert(config["exposure"]);
+  assert(config["auto_focus"]);
+  assert(config["auto_exposure"]);
+  assert(config["width_pixels"]);
+  assert(config["height_pixels"]);
+
+  CameraConfiguration camera_config;
+  {
+    camera_config.frames_per_second = config["frames_per_second"].as<int>();
```

Comment:
Can we shut down in a nice way if these fail to parse?

Reply:
I realize it's a nogo on crash anyway.

Reply:
Noble. Yes.